# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc/glibc23/in_divdi3_c.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


[see http://sources.redhat.com/ml/libc-alpha/2005-11/msg00033.html]

H.J.
----
2005-11-08  H.J. Lu  <hongjiu.lu@intel.com>

        * sysdeps/wordsize-32/Makefile (CPPFLAGS-divdi3.c): Set to
        -Din_divdi3_c

        * sysdeps/wordsize-32/symbol-hacks.h: Skip if in_divdi3_c is
        defined.

--- ./sysdeps/wordsize-32/Makefile.as     2002-09-18 23:47:56.000000000 -0700
+++ ./sysdeps/wordsize-32/Makefile        2005-11-07 15:13:38.000000000 -0800
@@ -2,5 +2,6 @@ ifeq ($(subdir),csu)
 ifeq (yes,$(build-shared))
 sysdep_routines += divdi3
 shared-only-routines += divdi3
+CPPFLAGS-divdi3.c = -Din_divdi3_c
 endif
 endif
--- ./sysdeps/wordsize-32/symbol-hacks.h.as       2004-03-08 12:59:34.000000000 -0800
+++ ./sysdeps/wordsize-32/symbol-hacks.h  2005-11-07 14:57:49.000000000 -0800
@@ -22,7 +22,7 @@
    therefore we get PLTs.  Unnecessarily so.  Changing gcc is a big
    task which might not be worth it so we play tricks with the
    assembler.  */
-#if !defined __ASSEMBLER__ && !defined NOT_IN_libc && defined SHARED
+#if !defined __ASSEMBLER__ && !defined in_divdi3_c && !defined NOT_IN_libc && defined SHARED
 asm ("__divdi3 = __divdi3_internal");
 asm ("__udivdi3 = __udivdi3_internal");
 asm ("__moddi3 = __moddi3_internal");
